src/Stratis.Features.FederatedPeg/Wallet/DeterministicCoinSelector.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg.Wallet
+{
+    /// <summary>
+    /// Based on the DefaultCoinSelector but with randomness removed, and GroupByScriptKey set to false.
+    /// </summary>
+    public class DeterministicCoinSelector : ICoinSelector
```

Comment:
Ahh, I should add tests to this class before we merge actually.

Reply:
Nice 💯

Reply:
Interesting idea to override coin selecotor, this only order UTXOs right?

Reply:
Yeah it chooses which UTXOs to use. The DefaultCoinSelector sometimes uses randomness to choose the outputs so I wanted to avoid that.